admin/spec/service/check-window.service.spec.js

```diff
@@ -189,5 +189,55 @@ describe('check-window.service', () => {
       expect(result.toString).toBe(expected.toString)
       expect(result).toBeTruthy()
     })
+    describe('#isLoginAllowed', () => {
+      it('should allow pupil to login if within range of active check window', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue({
+          checkStartDate: moment.utc().subtract(1, 'days'),
+          checkEndDate: moment.utc().add(1, 'days')
+        })
+        try {
+          await service.isLoginAllowed()
+        } catch (error) {
+          fail('not expected to throw')
+        }
+        done()
+      })
+      it('should disallow pupil to login if there is now active window', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue(Promise.resolve({}))
+        try {
+          await service.isLoginAllowed()
+          fail('not expected to throw')
+        } catch (error) {
+          expect(error.message).toBe('There is no active check window')
+        }
+        done()
+      })
+      it('should disallow pupil to login if checkStartDate is in the future', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue(Promise.resolve({
+          checkStartDate: moment.utc().add(1, 'days'),
+          checkEndDate: moment.utc().add(2, 'days')
+        }))
+        try {
+          await service.isLoginAllowed()
+          fail('not expected to throw')
+        } catch (error) {
+          expect(error.message).toBe('The active check window has not started')
```

Comment:
I find it odd to refer to the check window being "active" when it has not started.  Perhaps just "There is no open check window" would be clearer.   

Even more odd, is that `checkWindowDataService.sqlFindOneCurrent` would return a checkWindow that is not open, as I would expect it to return null| undefined if there wasn't a current active check Window.